File: src/main/java/org/droidmate/analyzer/ReportFormatter.java
package org.droidmate.analyzer;

import org.droidmate.analyzer.api.IApi;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;

/**
 * Created by nataniel on 19.05.17.
 */
public class ReportFormatter {
    private static SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
    static String formatApiList(List<IApi> apis){
        StringBuilder b = new StringBuilder();
        apis.forEach(p -> b.append(String.format("\t%s\n", p.toString())));

        return b.toString();
    }

    static String formatDate(Date date){
        return dateFormat.format(date);
    }
}

File: src/main/kotlin/org/droidmate/analyzer/ReportFormatter.kt
package org.droidmate.analyzer

import org.droidmate.analyzer.api.IApi
import java.text.SimpleDateFormat
import java.util.*

/**
 * Format template for the report
 */
object ReportFormatter {
    private val dateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS")

    internal fun formatApiList(apis: List<IApi>): String {
        val b = StringBuilder()
        apis.forEach { p -> b.append(String.format("\t%s\n", p.toString())) }

        return b.toString()
    }

    internal fun formatDate(date: Date): String {
        return dateFormat.format(date)
    }
}
